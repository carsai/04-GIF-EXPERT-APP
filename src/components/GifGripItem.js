import React from 'react';
import PropTypes from 'prop-types';

export const GifGripItem = ({title, url}) => {

    return (
        <div className='card animate__animated animate__bounce'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGripItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
