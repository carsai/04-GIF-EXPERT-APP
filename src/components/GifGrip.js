import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGripItem } from './GifGripItem';

export const GifGrip = ({category}) => {

    const {data, loading} = useFetchGifs(category);     

    return (
        <>
            <h3 className='animate__animated animate__bounce'>{category}</h3> 
            {loading && <p className='animate__animated animate__flash'>cargando</p>}
            <div className='card-grid'>
            {
                    data.map(img => <GifGripItem 
                            key={img.id} 
                            {...img}
                        />)
            }
            </div>
        </>
    )
}
