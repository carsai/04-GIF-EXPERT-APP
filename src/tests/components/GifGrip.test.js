import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';

import { GifGrip } from '../../components/GifGrip';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas con <GifGrip />', () => {
    
    const categoria = 'One Punch';    
    
    test('Debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrip category={categoria} />);
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('Debe de mostrar items cuando hay una imagen usando useFetchGifs', () => {

        const data = [
            {
                id: '1234567',
                title: 'titulo GIF',
                url: 'https://localhost:8080/quesito.gif'
            },
            {
                id: '987654',
                title: 'titulo GIF 2',
                url: 'https://localhost:8080/quesito2.gif'
            }
        ];
        
        useFetchGifs.mockReturnValue({
            data,
            loading: false
        });
        
        const wrapper = shallow(<GifGrip category={categoria} />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGripItem').length).toBe(data.length);
        
    });
    
});
