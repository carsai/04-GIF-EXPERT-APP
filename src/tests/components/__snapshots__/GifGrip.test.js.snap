// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifGrip /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
